### ISSUE TITLE
Error running 'make check'  from Source on Ubuntu

### ISSUE THREAD

**Issue by ribsthakkar** [moves-rwth/storm#102]
I am having an issue running the tests when installing from source on Ubuntu.
There seems to be issues with the CUDD 3rd party library
```
[  2%] Built target copy_storm_pomdp_headers
[  2%] Performing build step for 'sylvan'
-- sylvan build command succeeded.  See also /home/ribsthak/storm/build/sylvan/src/sylvan-stamp/sylvan-build-*.log
[  2%] No install step for 'sylvan'
[  2%] Completed 'sylvan'
[  2%] Built target sylvan
[  3%] Built target eigen_src
[  4%] Built target l3pp_ext
[  4%] Built target copy_storm_version_info_headers
[  4%] Built target storm-version-info
[  4%] Built target copy_resources_headers
[  4%] Performing build step for 'cudd3'
CMake Error at /home/ribsthak/storm/build/resources/3rdparty/cudd-3.0.0/src/cudd3-stamp/cudd3-build-RELEASE.cmake:49 (message):
  Command failed: 2

   'make' 'CFLAGS=-O3 -w -DPIC -DHAVE_IEEE_754 -fno-common -ffast-math -fno-finite-math-only -march=native'

  See also

    /home/ribsthak/storm/build/resources/3rdparty/cudd-3.0.0/src/cudd3-stamp/cudd3-build-*.log


make[3]: *** [CMakeFiles/cudd3.dir/build.make:112: resources/3rdparty/cudd-3.0.0/src/cudd3-stamp/cudd3-build] Error 1
make[2]: *** [CMakeFiles/Makefile2:1075: CMakeFiles/cudd3.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:681: CMakeFiles/check.dir/rule] Error 2
make: *** [Makefile:210: check] Error 2
```
The cudd-build-*.log files dont' show any issues:
```
configure: WARNING: Doxygen not found - continuing without Doxygen support
configure: WARNING: pdflatex not found - unable to compile manual to PDF
configure: WARNING: makeindex not found - unable to compile manual to PDF
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for gcc... /usr/bin/cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether /usr/bin/cc accepts -g... yes
checking for /usr/bin/cc option to accept ISO C89... none needed
checking whether /usr/bin/cc understands -c and -o together... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of /usr/bin/cc... gcc3
checking whether we are using the GNU C++ compiler... yes
checking whether /usr/bin/c++ accepts -g... yes
checking dependency style of /usr/bin/c++... gcc3
checking for ar... ar
checking the archiver (ar) interface... ar
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by /usr/bin/cc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from /usr/bin/cc object... ok
checking for sysroot... no
checking for mt... mt
checking if mt is a manifest tool... no
checking how to run the C preprocessor... /usr/bin/cc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if /usr/bin/cc supports -fno-rtti -fno-exceptions... no
checking for /usr/bin/cc option to produce PIC... -fPIC -DPIC
checking if /usr/bin/cc PIC flag -fPIC -DPIC works... yes
checking if /usr/bin/cc static flag -static works... yes
checking if /usr/bin/cc supports -c -o file.o... yes
checking if /usr/bin/cc supports -c -o file.o... (cached) yes
checking whether the /usr/bin/cc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking how to run the C++ preprocessor... /usr/bin/c++ -E
checking for ld used by /usr/bin/c++... /usr/bin/ld -m elf_x86_64
checking if the linker (/usr/bin/ld -m elf_x86_64) is GNU ld... yes
checking whether the /usr/bin/c++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking for /usr/bin/c++ option to produce PIC... -fPIC -DPIC
checking if /usr/bin/c++ PIC flag -fPIC -DPIC works... yes
checking if /usr/bin/c++ static flag -static works... yes
checking if /usr/bin/c++ supports -c -o file.o... yes
checking if /usr/bin/c++ supports -c -o file.o... (cached) yes
checking whether the /usr/bin/c++ linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether byte ordering is bigendian... no
checking for doxygen... no
checking for pdflatex... no
checking for makeindex... no
checking for library containing pow... -lm
checking for pthread_create in -lpthread... yes
checking for library containing WSAStartup... no
checking for library containing GetProcessMemoryInfo... no
checking float.h usability... yes
checking float.h presence... yes
checking for float.h... yes
checking for inttypes.h... (cached) yes
checking limits.h usability... yes
checking limits.h presence... yes
checking for limits.h... yes
checking stddef.h usability... yes
checking stddef.h presence... yes
checking for stddef.h... yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking assert.h usability... yes
checking assert.h presence... yes
checking for assert.h... yes
checking math.h usability... yes
checking math.h presence... yes
checking for math.h... yes
checking for unistd.h... (cached) yes
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking sys/times.h usability... yes
checking sys/times.h presence... yes
checking for sys/times.h... yes
checking sys/resource.h usability... yes
checking sys/resource.h presence... yes
checking for sys/resource.h... yes
checking sys/wait.h usability... yes
checking sys/wait.h presence... yes
checking for sys/wait.h... yes
checking for working C++ thread header... yes
checking for stdbool.h that conforms to C99... yes
checking for _Bool... yes
checking for inline... inline
checking for size_t... yes
checking for uint16_t... yes
checking for uint32_t... yes
checking for ptrdiff_t... yes
checking size of int... 4
checking size of long... 8
checking size of void *... 8
checking size of long double... 16
checking whether we are building for a Win32 host... no
checking whether enough of C++11 is supported... yes
checking for pow... yes
checking for sqrt... yes
checking for strchr... yes
checking for strstr... yes
checking for powl... yes
checking for gethostname... yes
checking for getrlimit... yes
checking for getrusage... yes
checking for sysconf... yes
checking for +Infinity (IEEE 754 floating point)... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating dddmp/exp/test1.sh
config.status: creating dddmp/exp/test2.sh
config.status: creating dddmp/exp/test3.sh
config.status: creating dddmp/exp/test4.sh
config.status: creating dddmp/exp/test5.sh
config.status: creating dddmp/exp/test6.sh
config.status: creating dddmp/exp/test7.sh
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing libtool commands
--------------------------------------------------
Configuration summary for cudd 3.0.0

Build system   : x86_64-unknown-linux-gnu
Host system    : x86_64-unknown-linux-gnu
Prefix         : '/home/ribsthak/storm/build/resources/3rdparty/cudd-3.0.0'
Compilers      : '/usr/bin/cc    -Wall -Wextra -g -O3'
               : '/usr/bin/c++    -Wall -Wextra -std=c++0x -g -O3'
Shared library : yes
 dddmp enabled : no
 obj enabled   : yes
--------------------------------------------------
```

**Comment by sjunges** [comment]
Hi,

1. are there more error or log files in

/home/ribsthak/storm/build/resources/3rdparty/cudd-3.0.0/src/cudd3-stamp/

2. just to be sure, does the same error occur if you just run make?

3. What is the output when you run cmake .. ?

**Comment by ribsthakkar** [comment]
Ah yes, there seem to be additional error lines in the /home/ribsthak/storm/build/resources/3rdparty/cudd-3.0.0/src/cudd3-stamp/cudd3-build-err.log
```
./libtool: line 614: syntax error near unexpected token `$'in\r''
'/libtool: line 614: `    case ${1} in
make[4]: *** [Makefile:1677: cudd/libcudd_la-cuddAddAbs.lo] Error 2
make[3]: *** [Makefile:1048: all] Error 2
```
The same error occured when I ran `make` , but I didn't have any errors when running `cmake ..`

This  may be related issue now that I think about it, but I was having bash parsing errors for many files first running `make check`,  so I used dos2unix to convert the line endings for the entire directory. Maybe that messed some stuff up.

**Comment by ribsthakkar** [comment]
It seems like the error in my previous comment also has to do with some line endings issue (hence the \r), but I can't seem to figure out which Makefile that is in

**Comment by sjunges** [comment]
I would suggest to go back to the situation that you have before running dos2unix and see what the error message is there. Share that error message and the steps that you took to get there. 
Also, how did you obtain the sources? Just git cloning? You could run cmake? What was its output? 

 We have many ubuntu users that do not have bash errors when trying to install, so let's see what your error message is and if and how that is related to storm :-)

**Comment by ribsthakkar** [comment]
Ok so, I did a fresh start by completely removing the directory. I executed the following steps (from the Getting Started Page) without errors:

1. I cloned the stable branch from git
2. Created a `build/` directory
3. I was able to run `cmake ..` from within the build directory

The error begins when I first run `make` inside the build directory. The output I see from the command is
```
ribsthak@DESKTOP-50S06IT:~/storm/build$ make
Scanning dependencies of target carl
[  0%] Creating directories for 'carl'
[  0%] No download step for 'carl'
[  0%] No patch step for 'carl'
[  0%] No update step for 'carl'
[  0%] No configure step for 'carl'
[  0%] Performing build step for 'carl'

-- carl build command succeeded.  See also /home/ribsthak/storm/build/carl-prefix/src/carl-stamp/carl-build-*.log
[  0%] Performing install step for 'carl'
-- carl install command succeeded.  See also /home/ribsthak/storm/build/carl-prefix/src/carl-stamp/carl-install-*.log
[  0%] Completed 'carl'
[  0%] Built target carl
Scanning dependencies of target cudd3
[  0%] Creating directories for 'cudd3'
[  0%] No download step for 'cudd3'
[  0%] Performing patch step for 'cudd3'
[  0%] No update step for 'cudd3'
[  0%] Performing configure step for 'cudd3'
CMake Error at /home/ribsthak/storm/build/resources/3rdparty/cudd-3.0.0/src/cudd3-stamp/cudd3-configure-RELEASE.cmake:49 (message):
  Command failed: 2

   '/home/ribsthak/storm/resources/3rdparty/cudd-3.0.0/configure' '--enable-shared' '--enable-obj' '--with-pic=yes' '--prefix=/home/ribsthak/storm/build/resources/3rdparty/cudd-3.0.0' '--libdir=/home/ribsthak/storm/build/resources/3rdparty/cudd-3.0.0/lib' 'CC=/usr/bin/cc' 'CXX=/usr/bin/c++'

  See also

    /home/ribsthak/storm/build/resources/3rdparty/cudd-3.0.0/src/cudd3-stamp/cudd3-configure-*.log


make[2]: *** [CMakeFiles/cudd3.dir/build.make:107: resources/3rdparty/cudd-3.0.0/src/cudd3-stamp/cudd3-configure] Error 1
make[1]: *** [CMakeFiles/Makefile2:1104: CMakeFiles/cudd3.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
ribsthak@DESKTOP-50S06IT:~/storm/build$ cat resources/3rdparty/cudd-3.0.0/src/cudd3-stamp/cudd3-*.log
configure: error: cannot run /bin/bash /home/ribsthak/storm/resources/3rdparty/cudd-3.0.0/build-aux/config.sub
```
So it seems like it was successfully able to build the carl dependency but it had issues with the cudd dependency.

**Comment by sjunges** [comment]
- What was the CMake output?
- What happens if you run /bin/bash /home/ribsthak/storm/resources/3rdparty/cudd-3.0.0/build-aux/config.sub

**Comment by ribsthakkar** [comment]
If I remember correctly, last time I tried to run the config.sub file by hand and that's when I noticed the line ending issue. Here it is again:
```
/home/ribsthak/storm/resources/3rdparty/cudd-3.0.0/build-aux/config.sub: line 4: $'\r': command not found
/home/ribsthak/storm/resources/3rdparty/cudd-3.0.0/build-aux/config.sub: line 6: $'\r': command not found
/home/ribsthak/storm/resources/3rdparty/cudd-3.0.0/build-aux/config.sub: line 26: $'\r': command not found
/home/ribsthak/storm/resources/3rdparty/cudd-3.0.0/build-aux/config.sub: line 27: $'\r': command not found
/home/ribsthak/storm/resources/3rdparty/cudd-3.0.0/build-aux/config.sub: line 34: $'\r': command not found
/home/ribsthak/storm/resources/3rdparty/cudd-3.0.0/build-aux/config.sub: line 37: $'\r': command not found
/home/ribsthak/storm/resources/3rdparty/cudd-3.0.0/build-aux/config.sub: line 45: $'\r': command not found
/home/ribsthak/storm/resources/3rdparty/cudd-3.0.0/build-aux/config.sub: line 52: $'\r': command not found
/home/ribsthak/storm/resources/3rdparty/cudd-3.0.0/build-aux/config.sub: line 54: $'\r': command not found
/home/ribsthak/storm/resources/3rdparty/cudd-3.0.0/build-aux/config.sub: line 67: $'\r': command not found
/home/ribsthak/storm/resources/3rdparty/cudd-3.0.0/build-aux/config.sub: line 75: $'\r': command not found
/home/ribsthak/storm/resources/3rdparty/cudd-3.0.0/build-aux/config.sub: line 78: $'\r': command not found
/home/ribsthak/storm/resources/3rdparty/cudd-3.0.0/build-aux/config.sub: line 81: syntax error near unexpected token `$'in\r''
'home/ribsthak/storm/resources/3rdparty/cudd-3.0.0/build-aux/config.sub: line 81: `  case $1 in
```
The CMake output looked ok to me (`cmake ..`):
```
-- The CXX compiler identification is GNU 9.3.0
-- The C compiler identification is GNU 9.3.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Storm - CMake install dir: /usr/local/lib/CMake/storm
-- Storm - Building RELEASE version.
-- Storm - Could not find ccache.
-- Storm - Detected operating system Linux.
-- Storm - Assuming extension for shared libraries: .so
-- Storm - Assuming extension for static libraries: .a
-- Storm - Build static libraries.
-- Storm - Enabling link-time optimizations.
-- Performing Test COMPILER_C_WORKS
-- Performing Test COMPILER_C_WORKS - Success
-- Performing Test COMPILER_CXX_WORKS
-- Performing Test COMPILER_CXX_WORKS - Success
-- Storm - Using compiler configuration gcc 9.3.0.
-- Storm - Including Eigen 3.3.7.
-- Storm - Using boost 107100 (library version 1_71).
-- Storm - Including ExprTk.
-- Storm - Including Sparsepp.
-- Storm - Including ModernJSON.
-- Storm - Linking with Z3 (version 4.8.7). (Z3 version supports optimization)
-- Storm - Using system version of glpk.
-- Storm - Linking with glpk 4.65
-- Storm - Linking with CUDD 3.0.0.
-- Carl - Start of config process
-- -- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Carl - Storm 3rdparty binary dir: /home/ribsthak/storm/build/resources/3rdparty
-- Carl - Finished configuration.
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ribsthak/storm/build/resources/3rdparty/carl_download

Cloning into 'source_dir'...
Switched to a new branch 'master14'
-- Scanning dependencies of target carl-config
[ 12%] Creating directories for 'carl-config'
[ 25%] Performing download step (git clone) for 'carl-config'
Branch 'master14' set up to track remote branch 'master14' from 'origin'.
[ 37%] No patch step for 'carl-config'
[ 50%] Performing update step for 'carl-config'
Current branch master14 is up to date.
[ 62%] Performing configure step for 'carl-config'
-- The CXX compiler identification is GNU 9.3.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Version: carl 14.20
-- Using g++ 9.3.0
-- CXX Flags:  -std=c++14 -Wunknown-pragmas -pthread -fmax-errors=5 -fdiagnostics-color=auto
-- CXX Debug Flags: -g -O1
-- CXX Release Flags: -O3 -DNDEBUG -O3
-- Build type: RELEASE
-- Building static: no
-- cotire is disabled
-- GMP / GMPXX 6.2.0 was found at /usr/include/x86_64-linux-gnu and /usr/lib/x86_64-linux-gnu/libgmp.so
-- Boost 107100 was found at /usr/include and Boost_filesystem_SHARED;Boost_system_SHARED;Boost_program_options_SHARED;Boost_regex_SHARED;Boost_unit_test_framework_SHARED
-- Eigen3 3.3.7 was found at /usr/include/eigen3
-- Bliss is disabled
-- CLN 1.3.6 was found at /usr/include and /usr/lib/x86_64-linux-gnu/libcln.so
-- CoCoA is disabled
-- GiNaC 1.7.8 was found at /usr/include/ginac and /usr/lib/libginac.so
-- GTest 1.8.0 was found at /home/ribsthak/storm/build/resources/3rdparty/carl/resources/src/GTest-EP/googletest/include and /home/ribsthak/storm/build/resources/3rdparty/carl/resources/src/GTest-EP-build/googlemock/gtest/libgtest_main.a
-- Doxygen was not found.
-- Did not find clang-tidy, target tidy is disabled.
-- Did not find clang-format, target format is disabled.
-- Found PythonInterp: /usr/bin/python3.8 (found version "3.8.2")
-- Registered with cmake
-- Run "make .travis.yml" to rebuild the travis file.
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ribsthak/storm/build/resources/3rdparty/carl
[ 75%] No build step for 'carl-config'
[ 87%] No install step for 'carl-config'
[100%] Completed 'carl-config'
[100%] Built target carl-config

-- Carl - End of config process
-- Storm - Using shipped version of carl.
-- Storm - Linking with shipped carl 14.20 (include: /home/ribsthak/storm/build/resources/3rdparty/carl/include/, library /home/ribsthak/storm/build/resources/3rdparty/carl/lib/libcarl.so, CARL_USE_CLN_NUMBERS: ON, CARL_USE_GINAC: ON).
-- Storm - Use system version of xerces.
-- Storm (GSPN) - Linking with Xerces-c 3.2.2: /usr/lib/x86_64-linux-gnu/libxerces-c.so
-- Storm - Using shipped version of sylvan.
-- Storm - Linking with sylvan.
-- Storm - Linking with hwloc .
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE)
-- Storm - Version is 1.6.4 (dev) (version 1.6.3 + 1 commits), building from git: 41ae9d56244308529f43bc9759f023e9c5d49433 (dirty: DirtyState::Clean).
-- Registered with cmake
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ribsthak/storm/build
```

**Comment by ribsthakkar** [comment]
Ok I figured out the issue. I was cloning the repo in WSL but for some reason it still thought I was in Windows and was automatically applying the CRLF line endings. I need to check my git settings.... When I used wget to get the zip and run the installation, it works successfully.
